service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ImmunizationObservationStatementMapper.java

```
@@ -223,4 +228,27 @@ private String extractVaccinationProtocolString(Immunization.ImmunizationVaccina
 
         return vaccinationProtocol + VACCINATION_TARGET_DISEASE + targetDiseases;
     }
+
+    private String buildCode(Immunization immunization) {
+        var vaccinationProcedure = ExtensionMappingUtils.filterExtensionByUrl(immunization, VACCINATION_PROCEDURE_URL);
+        if (vaccinationProcedure.isPresent()) {
+            CodeableConcept codeableConcept = (CodeableConcept) vaccinationProcedure.get().getValue();
+            return codeableConceptCdMapper.mapCodeableConceptToCd(codeableConcept);
+        }
+        return StringUtils.EMPTY;
```

The vaccinationProcedure extension is required. Should throw error is missing.